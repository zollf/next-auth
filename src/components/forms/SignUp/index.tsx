import React, { useState } from 'react';
import Router from 'next/router';
import { Button, Input } from '@/components';

const SignUp = () => {
  const [name, setName] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState(false);

  const onSubmit = async () => {
    if (name && password) {
      try {
        const response = await fetch('/api/signUp', {
          method: 'POST',
          body: JSON.stringify({ name, password }),
        }).then((res) => res.json());
        if (!response.error) {
          setError(false);
          Router.push('/');
        } else {
          setError(true);
        }
      } catch (e) {
        setError(true);
      }
    } else {
      setError(true);
    }
  };

  return (
    <fieldset data-test-error={error}>
      {error && <p>An Error has occurred</p>}
      <section>
        <label htmlFor="username">Username</label>
        <Input value={name} name="name" type="text" onChange={(e) => setName(e.target.value)} />
      </section>
      <section>
        <label htmlFor="password">Password</label>
        <Input value={password} name="password" type="password" onChange={(e) => setPassword(e.target.value)} />
      </section>
      <Button onClick={onSubmit}>Submit</Button>
    </fieldset>
  );
};

export default SignUp;
