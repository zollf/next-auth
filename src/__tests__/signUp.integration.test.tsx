import React from 'react';
import { ReactWrapper, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import SignUp from '@/pages/signUp';

const component = <SignUp />;

describe('<SignUp />', () => {
  let wrapper!: ReactWrapper;
  beforeEach(() => {
    wrapper = mount(component);
  });

  it('matches its snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
