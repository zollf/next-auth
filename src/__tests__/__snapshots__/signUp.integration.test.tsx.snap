// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> matches its snapshot 1`] = `
<SignUp>
  <Wrapper>
    <div
      className="wrapper"
    >
      <h1>
        Sign Up
      </h1>
      <SignUp>
        <fieldset
          data-test-error={false}
        >
          <section>
            <label
              htmlFor="username"
            >
              Username
            </label>
            <Input
              name="name"
              onChange={[Function]}
              type="text"
              value=""
            >
              <input
                className="input"
                name="name"
                onChange={[Function]}
                type="text"
                value=""
              />
            </Input>
          </section>
          <section>
            <label
              htmlFor="password"
            >
              Password
            </label>
            <Input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            >
              <input
                className="input"
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </Input>
          </section>
          <Button
            onClick={[Function]}
          >
            <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              Submit
            </button>
          </Button>
        </fieldset>
      </SignUp>
    </div>
  </Wrapper>
</SignUp>
`;
